src/ophys_etl/transforms/roi_transforms.py

```diff
@@ -127,3 +129,148 @@ def crop_roi_mask(roi_mask: coo_matrix) -> coo_matrix:
     cropped_mask = roi_mask.tocsr()[min_row:max_row, min_col:max_col]
 
     return cropped_mask.tocoo()
+
+
+def coo_rois_to_lims_compatible(coo_masks: List[coo_matrix],
+                                max_correction_vals: motion_border,
+                                movie_shape: Tuple[int, int]
+                                ) -> List[Dict[str, Any]]:
+    """
+    Converts coo formatted ROIs to lims compatible format.
+    Parameters
+    ----------
+    coo_masks: List[coo_matrix]
+        A list of scipy coo_matrices representing ROI masks, each element of
+        list is a unique ROI.
+    max_correction_vals: motion_border
+        The max motion correction values identified in the motion correction
+        step of ophys segmentation pipeline
+    movie_shape: Tuple[int, int]
+        The frame shape of the movie from which ROIs were extracted in order
+        of: (height, width).
+
+    Returns
+    -------
+    List[Dict[str, Any]]
+        A list of dictionaries representing the ROIs in lims compatible
+        format, the data contained inside each one is as follows:
+        {
+            id: int
+            x: int
+            y: int
+            width: int
+            height: int
+            valid_roi: Bool
+            mask_matrix: List[List[Bool]]
+            max_correction_up: int
+            max_correction_down: int
+            max_correction_left: int
+            max_correction_right: int
+            mask_image_plane: int
+            exclusion_labels: List[int] codes are defined in 2019 Ophys Docs
+        }
+        For details about specific values see design document for 2020 Ophys
+        Segmentation Refactor and Update
+        """
+    compatible_rois = []
+    for temp_id, coo_mask in enumerate(coo_masks):
+        compatible_roi = _coo_mask_to_LIMS_compatible_format(coo_mask)
+        compatible_roi['id'] = temp_id  # popped off writing to LIMs
+        compatible_roi['cell_specimen_id'] = temp_id
+        compatible_roi['max_correction_up'] = max_correction_vals.up
+        compatible_roi['max_correction_down'] = max_correction_vals.down
+        compatible_roi['max_correction_right'] = max_correction_vals.right
+        compatible_roi['max_correction_left'] = max_correction_vals.left
+        compatible_roi['mask_image_plane'] = 0
+        compatible_roi['exclusion_labels'], compatible_roi['valid_roi'] = \
+            _check_exclusion(compatible_roi, movie_shape)
+        compatible_rois.append(compatible_roi)
+    return compatible_rois
+
+
+def _coo_mask_to_LIMS_compatible_format(coo_mask: coo_matrix) -> Dict:
```

Comment:
It appears the values are transposed. Make sure you add an asymmetrical test case.

Here is my example:
```
 mat = coo_matrix(np.array([[0, 1, 0, 1], [0, 1, 1, 0], [1, 0, 0, 1]]))  
_coo_mask_to_LIMS_compatible_format(mat)
```
Returns
```
{'x': 0,
 'y': 1,
 'width': 4,
 'height': 3,
 'mask_matrix': [[False, True, False, True],
  [False, True, True, False],
  [True, False, False, True]]}
```
But 'x' should be 0 and 'y' should be 1.

Double checked with the AllenSDK:
```
from allensdk.brain_observatory.roi_masks import create_roi_mask 
roi = create_roi_mask(4, 3, [0,0,0,0], roi_mask=mat.toarray())
roi.x    # returns 1
roi.y    # returns 0
``` 

Array format of the test:
```
array([
       [0, 0, 1, 0, 1],
       [0, 0, 1, 1, 0],
       [0, 1, 0, 0, 1]
])
```

Reply:
The masks weren't transposed the x and y were flipped in the function, I fixed the flipping and added a test case for this behavior.

Reply:
Yes, the 'x' and 'y' keys were transposed as I said.